conans/client/api/conan_api.py

```diff
@@ -90,19 +166,81 @@ def user_update(self, user_name, user_pasword):
         return {}
 
     @api_method
-    def search_recipes(self, query, remote_patterns=None, local_cache=False):
-        remote = None
-        if remote_patterns is not None and len(remote_patterns) > 0:
-            remote = remote_patterns[0].replace("*", "remote")
+    def search_local_recipes(self, query):
+        data = {
+            "remote": None,
+            "error": False,
+            "results": None
+        }
+
+        remotes = self.app.cache.registry.load_remotes()
+        search = Search(self.app.cache, self.app.remote_manager, remotes)
+
+        try:
+            references = search.search_local_recipes(query)
+        except ConanException as exc:
+            data["error"] = True
+            raise
+
+        results = []
+        for reference in references:
+            reference = repr(reference)
+            result = {
+                "name": get_recipe_name(reference),
```

Comment:
Maybe this is not necessary, if you can do ``name = reference.name`` before converting to string. ``get_recipe_name()`` should be unnecessary function.

Reply:
You're absolutely right. Fixed.